cdap-docs/admin-manual/source/upgrading/cloudera.rst

```diff
@@ -0,0 +1,108 @@
+noted.
+
+The following is the generic procedure for Major/Minor version upgrades:
+
+#. Stop all flows, services, and other programs in all your applications.
+
+#. Stop all CDAP services.
+
+#. Ensure your installed version of the CSD matches the target version of CDAP. For
+   example, CSD version 3.0.* is compatible with CDAP version 3.0.*.  Download `the latest
+   version of the CSD <http://cask.co/resources/#cdap-integrations>`__.
+
+#. Use the Cloudera Manager UI to download, distribute, and activate the corresponding
+   CDAP parcel version on all cluster hosts.
+
+#. Before starting services, run the *CDAP Upgrade Tool* to upgrade CDAP. From the CDAP Service 
+   page, select "Run CDAP Upgrade Tool" from the Actions menu.
+
+#. Start the CDAP services.  At this point it may be necessary to correct for any changes in
+   the CSD.  For example, if new CDAP services were added or removed, you must add or
+   remove role instances as necessary. Check the :ref:`release-specific upgrade notes
+   <cloudera-release-specific-upgrade-notes>` below for any specific instructions.
+   
+#. After CDAP services have started, run the *CDAP Post-Upgrade Tasks* to perform any necessary
+   upgrade steps against the running services.  From the CDAP Service page, select "Run CDAP
+   Post-Upgrade Tasks."
+
+#. To upgrade existing ETL applications created using the 3.2.x versions of ``cdap-etl-batch`` or 
+   ``cdap-etl-realtime``, there is are :ref:`separate instructions on doing so <cdap-apps-etl-upgrade>`.
+
+#. You must recompile and then redeploy your applications prior to using them.
+
+
+Upgrading CDH
+=============
+
+.. _cloudera-release-specific-upgrade-notes:
+
+These steps cover what to do when upgrading the version of CDH of an existing CDAP installation.
+As the different versions of CDH can use different versions of HBase, upgrading from
+one version to the next can require that the HBase coprocessors be upgraded to the correct
+version. The steps below will, if required, update the coprocessors appropriately.
+
+**It is important to perform these steps as described, otherwise the coprocessors may not
+get upgraded correctly and HBase regionservers may crash.** In the case where something
+goes wrong, see these troubleshooting instructions for :ref:`problems while upgrading CDH
+<faqs-cloudera-troubleshooting-upgrade-cdh>`.
+
+**Upgrade Steps**
+
+1. Upgrade CDAP to a version that will support the new CDH version, following the usual
+   :ref:`CDAP-Cloudera Manager upgrade procedure <admin-upgrading-cloudera-cdap>`. 
+
+#. After upgrading CDAP, start CDAP and check that it is working correctly.
+
+#. Using Cloudera Manager (CM), stop all CDAP application and services.
+
+#. Disable all CDAP tables; from an HBase shell, run the command::
+
+    > disable_all 'cdap.*'
+    
+#. Upgrade to the new version of CDH.
+
+#. Stop all CDAP services, as CM will have (again) auto-started CDAP.
```

Comment:
"CM may have".  It does restart when using the automatic upgrade option, but there is a manual option too which I presume (not sure) gives you control to upgrade/restart each service individually.